AppCenter/AppCenterTests/MSLogDBStorageTests.m

```diff
@@ -622,36 +625,168 @@ - (void)testAddLogsDoesNotExceedCapacity {
   }
 }
 
-- (void)testOldestLogsAreDeletedFirstWhenCapacityIsReached {
+- (void)testOldestNormalPriorityLogsAreDeletedFirstWhenCapacityIsReachedAndNewNormalPriorityLogComes {
 
   // If
   long maxCapacityInBytes = kMSTestStorageSizeMinimumUpperLimitInBytes + kMSDefaultPageSizeInBytes;
-  NSArray *addedLogs = [self fillDatabaseWithLogsOfSizeInBytes:maxCapacityInBytes];
-  MSAbstractLog *firstLog = addedLogs[0];
-  int initialLogCount = (int)[addedLogs count];
-  __block int originalLogsCount = initialLogCount;
+  NSArray *addedDbIds = [self fillDatabaseWithLogsOfSizeInBytes:maxCapacityInBytes ofPriority:MSFlagsPersistenceNormal];
+  NSNumber *firstLogDbId = addedDbIds[0];
 
   // When
   [self.sut setMaxStorageSize:maxCapacityInBytes
             completionHandler:^(__unused BOOL success){
             }];
-  while (originalLogsCount < initialLogCount) {
-    MSAbstractLog *additionalLog = [MSAbstractLog new];
-    additionalLog.sid = MS_UUID_STRING;
-    BOOL logSavedSuccessfully = [self.sut saveLog:additionalLog withGroupId:kMSAnotherTestGroupId flags:MSFlagsDefault];
-    NSString *originalLogsFilter = [NSString stringWithFormat:@"\"%@\" = '%@'", kMSGroupIdColumnName, kMSTestGroupId];
-    NSArray<id<MSLog>> *originalLogs = [self loadLogsWhere:originalLogsFilter];
-    originalLogsCount = (int)[originalLogs count];
-    if (originalLogsCount < initialLogCount) {
-      XCTAssertEqual(originalLogsCount, initialLogCount - 1);
-      BOOL containsFirstLog = [self logs:originalLogs containLogWithSessionId:firstLog.sid];
-      XCTAssertFalse(containsFirstLog);
-    }
+  MSAbstractLog *additionalLog = [MSAbstractLog new];
+  BOOL logSavedSuccessfully = [self.sut saveLog:additionalLog withGroupId:kMSAnotherTestGroupId flags:MSFlagsPersistenceNormal];
 
-    // Then
-    XCTAssertTrue([self.storageTestUtil getDataLengthInBytes] <= maxCapacityInBytes);
-    XCTAssertTrue(logSavedSuccessfully);
+  // Then
+  XCTAssertTrue([self.storageTestUtil getDataLengthInBytes] <= maxCapacityInBytes);
+  XCTAssertTrue(logSavedSuccessfully);
+  XCTAssertFalse([self containsLogWithDbId:firstLogDbId]);
+
+  NSString *whereCondition = [NSString stringWithFormat:@"\"%@\" = '%@'", kMSGroupIdColumnName, kMSAnotherTestGroupId];
+  NSArray<id<MSLog>> *loadedLogs = [self loadLogsWhere:whereCondition];
+  XCTAssertEqual(loadedLogs.count, 1);
+  XCTAssertEqualObjects(loadedLogs[0].sid, additionalLog.sid);
+  XCTAssertEqual(1, [self findUnknownDBIdsFromKnownIdList:addedDbIds].count);
+}
+
+- (void)testOldestNormalPriorityLogsAreDeletedFirstWhenCapacityIsReachedAndNewCriticalPriorityLogComes {
```

Comment:
This is hard to read, I would suggest something shorter such as `putCriticalPriorityLogPurgesOldestNormalPriorityLogs`.